WangscapeTest/TestPowInt.cpp

```diff
@@ -0,0 +1,107 @@
+        one = 1;
+        two = 2;
+        three = 3;
+        five = 5;
+        max = std::numeric_limits<I>::max();
+        digits = static_cast<I>(std::numeric_limits<I>::digits);
+
+    };
+    ~TestPowInt() {};
+};
+
+typedef ::testing::Types<
+    char, short, int, long, long long,
+    unsigned char, unsigned short, unsigned int, unsigned long, unsigned long long>
+    IntTypes;
+TYPED_TEST_CASE(TestPowInt, IntTypes);
+
+TYPED_TEST(TestPowInt, TestPowInt00)
+{
+    EXPECT_EQ(this->one, pow(this->zero, this->zero));
+}
+
+TYPED_TEST(TestPowInt, TestPowIntX0)
+{
+    EXPECT_EQ(this->one, pow(this->one, this->zero));
+    EXPECT_EQ(this->one, pow(this->two, this->zero));
+    EXPECT_EQ(this->one, pow(this->three, this->zero));
+    EXPECT_EQ(this->one, pow(this->five, this->zero));
+    EXPECT_EQ(this->one, pow(this->digits, this->zero));
+    EXPECT_EQ(this->one, pow(this->max, this->zero));
+}
+
+TYPED_TEST(TestPowInt, TestPowInt0X)
+{
+    EXPECT_EQ(this->zero, pow(this->zero, this->one));
+    EXPECT_EQ(this->zero, pow(this->zero, this->two));
+    EXPECT_EQ(this->zero, pow(this->zero, this->three));
+    EXPECT_EQ(this->zero, pow(this->zero, this->five));
+    EXPECT_EQ(this->zero, pow(this->zero, this->digits));
+    EXPECT_EQ(this->zero, pow(this->zero, this->max));
+}
+
+TYPED_TEST(TestPowInt, TestPowIntX1)
+{
+    EXPECT_EQ(this->one, pow(this->one, this->one));
+    EXPECT_EQ(this->two, pow(this->two, this->one));
+    EXPECT_EQ(this->three, pow(this->three, this->one));
+    EXPECT_EQ(this->five, pow(this->five, this->one));
+    EXPECT_EQ(digits, pow(this->digits, this->one));
+    EXPECT_EQ(max, pow(max, this->one));
+}
+
+TYPED_TEST(TestPowInt, TestPowInt1X)
+{
+    EXPECT_EQ(this->one, pow(this->one, this->one));
+    EXPECT_EQ(this->one, pow(this->one, this->two));
+    EXPECT_EQ(this->one, pow(this->one, this->three));
+    EXPECT_EQ(this->one, pow(this->one, this->five));
+    EXPECT_EQ(this->one, pow(this->one, this->digits));
+    EXPECT_EQ(this->one, pow(this->one, this->max));
+}
+
+TYPED_TEST(TestPowInt, TestPowIntExamples)
+{
+    EXPECT_EQ(4, pow(this->two, this->two));
+    EXPECT_EQ(9, pow(this->three, this->two));
+    EXPECT_EQ(25, pow(this->five, this->two));
+    EXPECT_EQ(this->one << (this->digits - 1), pow(this->two, this->digits - 1));
+
+    EXPECT_EQ(8, pow(this->two, this->three));
+    EXPECT_EQ(27, pow(this->three, this->three));
+    EXPECT_EQ(125, pow(this->five, this->three));
+
+    EXPECT_EQ(32, pow(this->two, this->five));
+}
+
+TYPED_TEST(TestPowInt, TestPowIntTooLarge)
+{
+    EXPECT_ANY_THROW(pow(this->two, this->digits));
+    EXPECT_ANY_THROW(pow(this->two, this->max));
+    EXPECT_ANY_THROW(pow(this->three, this->digits));
+    EXPECT_ANY_THROW(pow(this->three, this->max));
+    EXPECT_ANY_THROW(pow(this->digits, this->digits));
+    EXPECT_ANY_THROW(pow(this->digits, this->max));
+    EXPECT_ANY_THROW(pow(this->max, this->digits));
+    EXPECT_ANY_THROW(pow(this->max, this->max));
+}
```

Comment:
`digits**digits` and `digits**max` can be removed.

While rewriting the conditionals used in `pow`, I broke `EXPECT_ANY_THROW` for `max**digits` and `max**max`, but not when `base` was 2, 3, or `digits`. A pretty clear demonstration of the value of those two tests.

With tests like [TestLinearMovingScaleBiasShort](https://github.com/Wangscape/Wangscape/blob/master/WangscapeTest/TestMovingScaleBias.cpp#L57) I would agree with you that extra tests bring too much maintenance work; each test assertion requires some visualisation and mental arithmetic to verify. However, with test targets as simple as `pow` and test assertions as simple as "an error should be thrown when `base >= 2` and `exp >= digits`", I don't think that's really a valid concern.